docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
+       // You can define your own function for setting a dynamic context
+       // or provide a static value
+      return getDynamicContext(ctx, msg, args);
+    },
   },
-});
+  wsServer,
+);
 ```
 
-These functions are passed the following parameters:
-
-<table class="field-table">
-  <thead>
-    <tr>
-      <th>Name /<br/>Type</th>
-      <th>Description</th>
-    </tr>
-  </thead>
+Now here is where things might get a little bit confusing, notice above that the `context` property's first argument is `ctx`, also known as context. _What?_ The [`Context`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.Context.md) type in `graphql-ws` is **not the same thing** as the GraphQL `context` argument passed to your resolvers.
 
-<tbody>
-<tr>
-<td>
+The `Context` type from `graphql-ws` is where information about your subscription server and its requests resides. You can access incoming parameters from a client `subscription` request by checking the `Context.connectionParams`.
 
-###### `connectionParams`
+Below is an example of the common use case of extracting an authentication token from a client `subscription` request and using it to find the current user:
 
-`Object`
-
-</td>
-<td>
-
-**Passed to `onConnect` only.**
-
-An object containing parameters included in the request, such as an [authentication token](#example-authentication-with-onconnect).
-
-For details, see [Authenticate over WebSocket](https://www.apollographql.com/docs/react/data/subscriptions/#4-authenticate-over-websocket-optional) in the Apollo Client documentation.
-
-</td>
-</tr>
-
-<tr>
-<td>
-
-###### `webSocket`
-
-`WebSocket`
-</td>
-<td>
-
-The connecting or disconnecting [`WebSocket`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket).
+```ts
+const getDynamicContext = async (ctx, msg, args) => {
+  // ctx is the graphql-ws Context where connectionParams live
+ if (ctx.connectionParams.authentication) {
+    const currentUser = await findUser(connectionParams.authentication);
+    return { currentUser };
+  }
+  // Otherwise let our resolvers know we don't have a current user
+  return { currentUser: null };
+};
 
-</td>
-</tr>
+useServer(
+  {
+    schema,
+    // Adding a context property lets you add data to your GraphQL operation context
+     context: (ctx, msg, args) => {
+       // Returning an object will add that information to our
+       // GraphQL context, which all of our resolvers have access to.
+      return getDynamicContext(ctx, msg, args);
+    },
+  },
+  wsServer,
+);
+```
+Putting it all together, if the `useServer.context` function returns an _object_ that object is passed to the GraphQL operation `context`, which is available to your resolvers.
 
-<tr>
-<td>
+Note that the `context` option is called once per `subscription` request, not once per event emission. Meaning that in the above example, every time a client sends a `subscription` operation we would check their authentication token.
```

Comment:
"Meaning" -> "This means"

"operation we would check" -> "operation, we check"